jerry-ext/include/jerryscript-ext/context.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_CONTEXT_H
+#define JERRYX_CONTEXT_H
+
+#include "jerryscript.h"
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its second parameter
+ */
+void *jerryx_context_init (void);
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its third parameter
+ */
+void jerryx_context_deinit (void *user_context_p);
+
+typedef struct
+{
+  /** the initialization function */
+  jerry_user_context_init_cb init_cb;
+  /** the deinitialization function */
+  jerry_user_context_deinit_cb deinit_cb;
+} jerryx_context_slot_t;
+
+#ifdef __MACH__
+#define JERRYX_CONTEXT_SECTION "__DATA,jerryx_ctx_slots"
+#else /* !__MACH__ */
+#define JERRYX_CONTEXT_SECTION "jerryx_ctx_slots"
+#endif /* __MACH__ */
+
+#define JERRYX_CONTEXT_DEFINE_SLOT(slot_name, init_cb_ptr, deinit_cb_ptr)         \
+static jerryx_context_slot_t __jerryx_context_slot_ ## slot_name                  \
+__attribute__((used, section(JERRYX_CONTEXT_SECTION), aligned(sizeof(void *)))) = \
+{                                                                                 \
+  .init_cb = (init_cb_ptr),                                                       \
+  .deinit_cb = (deinit_cb_ptr)                                                    \
+};
```

Comment:
hm, you sure about this? the linked code _does_ make use of semicolons. `NODE_MODULE_X` is used in the definition of `NODE_MODULE` (same file, line 494), and if you search for `NODE_MODULE` in the repository, you can see it used with a trailing semicolon (https://github.com/nodejs/node/blob/db1087c9757c31a82c50a1eba368d8cba95b57d0/benchmark/misc/function_call/binding.cc)